modules/mapbox-layers/README.md

```diff
@@ -2,26 +2,90 @@
 
 Enables the use deck.gl layers as custom mapbox layers, enabling seamless interleaving of mapbox and deck.gl layers.
 
-See [deck.gl](http://deck.gl) for documentation.
+To create a mapbox-compatible deck.gl layer:
+
+```js
+import DeckLayer from '@deck.gl/mapbox-layers';
+import {ScatterplotLayer} from '@deck.gl/core-layers';
+
+const myDeckLayer = new DeckLayer({
+    id: 'my-scatterplot',
+    type: ScatterplotLayer,
+    data: [
+        {position: [-74.5, 40], size: 100}
+    ],
+    getPosition: d => d.position,
+    getRadius: d => d.size,
+    getColor: [255, 0, 0]
+});
+```
+
+To add the layer to mapbox:
+
+```js
+import mapboxgl from 'mapbox-gl';
+
+mapboxgl.accessToken = '<your access token here>';
+var map = new mapboxgl.Map({
+    container: 'map',
+    style: 'mapbox://styles/mapbox/streets-v9',
+    center: [-74.50, 40],
+    zoom: 9
+});
+
+map.addLayer(myDeckLayer);
+```
+
+See [deck.gl](http://deck.gl) for documentations and examples on how to create layers.
 
 
 ## DeckLayer
 
-The `DeckLayer` is a custom mapbox layer class that renders a list of deck.gl layers inside the mapbox canvas / WebGL context. This is in contrast to the typical deck.gl/mapbox integration where the deck.gl layers are rendered separately
+The `DeckLayer` is a custom mapbox layer class that renders a deck.gl layer inside the mapbox canvas / WebGL context. This is in contrast to the typical deck.gl/mapbox integration where the deck.gl layers are rendered into a separate canvas.
+
+##### constructor
+
+```
+import DeckLayer from '@deck.gl/mapbox-layers';
+
+new DeckLayer(props);
+```
+
+Parameters:
+
+- `props` (Object)
+    + `props.id` (String) - an unique id is required for each layer.
+    + `props.type` (`Layer`) - a class that extends deck.gl's base `Layer` class.
+    + Any other prop needed by this layer, specified by `type`. See deck.gl's [layer catalog](http://deck.gl/#/documentation/deckgl-api-reference/layers/layer) for each layer's props.
+
+##### setProps(props)
+
+```js
+const layer = new DeckLayer({
+    id: 'my-scatterplot',
+    type: ScatterplotLayer,
+    ...
+});
+
+layer.setProps({
+    radiusScale: 2
+});
+```
+
+Update a layer after it's added.
 
 
 ## Advantages and Limitations
 
-Advantages:
+### Advantages
 
-* mapbox and deck.gl layers can be freely "interleaved", enabling a number of important uses cases as described below.
+* mapbox and deck.gl layers can be freely "interleaved", enabling a number of layer mixing effects, such as drawing behind map labels; z-occlusion between deck.gl 3D objects and Mapbox buildings; etc.
 * mapbox and deck.gl will share a single canvas and WebGL context, saving system resources.
 
-Disadvantages:
+### Limitations
```

Comment:
* Only works with mapbox-gl API
* No react-map-gl integration...
* deck.gl API cannot be used

Reply:
It can technically be used with react-map-gl, by accessing the map instance. We could also add `addLayer` and `removeLayer` methods onto the React component.

Reply:
Yes... I suppose we have three cases then:

1. Use as a "mapbox" layer pack with raw mapbox API (supported)
2. Use as a "mapbox" layer pack with `react-map-gl` (can be supported with small extensions to react-map-gl, as you suggest).
3. Use with `DeckGL` to let deck layers be rendered in between mapbox layers (not supported).

I think it would be great to add support for #2 especially as effort seems low, but obviously not having support for #3 is going to be a blocker for many users.